src/checkbox/src/Checkbox.js

```diff
@@ -119,10 +147,11 @@ class Checkbox extends PureComponent {
           type="checkbox"
           name={name}
           value={value}
-          checked={checked}
+          checked={checked || indeterminate}
           onChange={onChange}
           disabled={disabled}
           {...(isInvalid ? { 'aria-invalid': true } : {})}
+          ref={this.setIndeterminate}
```

Comment:
`ref` is actually not available on `Box` since it's a React component — instead `Box` uses the non-standard but conventional `innerRef` prop. 

Also, this might actually not work completely since I think `ref/innerRef` is only called once in the lifecycle of the component. We might need to opt-in to `componentDidUpdate`: https://reactjs.org/docs/react-component.html#componentdidupdate